framework/src/utils/MooseUtils.C

```diff
@@ -561,6 +564,46 @@ toUpper(const std::string & name)
   return upper;
 }
 
+ExecFlagEnum
+getDefaultExecFlagEnum()
+{
+  ExecFlagEnum exec_enum = ExecFlagEnum();
+  exec_enum.addAvailableFlags(EXEC_NONE,
+                              EXEC_INITIAL,
+                              EXEC_LINEAR,
+                              EXEC_NONLINEAR,
+                              EXEC_TIMESTEP_END,
+                              EXEC_TIMESTEP_BEGIN,
+                              EXEC_FINAL,
+                              EXEC_CUSTOM);
+  return exec_enum;
+}
+
+template <typename T>
+T
+stringToNumber(const std::string & input)
+{
+  T output;
+  std::istringstream ss(input);
+  if (!(ss >> output) || !ss.eof())
+    mooseError(
```

Comment:
Well this isn't going to work for the general case. We routinely allow this conversion to fail and are fine with that. Every time you supply a boundary or block like "left" for instance. If this utility is going to be used, you'll have to throw an exception here because you can't presume that users will always want to fail. Also, we have many different error messages that are more specific than this one.

Reply:
What would be a good alternative without doing the conversion twice? Could this return a std::pair<T, bool>, where the bool provides pass/fail result?

Reply:
You could return a pair, but what would be the value of T when the conversion fails, undefined? I'd just throw, that's cleaner. You don't have to catch, you can just let the error make MOOSE crash.